### DISCUSSION TITLE
Api standards for front-end templates

### DISCUSSION THREAD

**Discussion by jonhurrell** [dwp/design-system#221] [Q&A] [answered]
#### Context
From discussions with @htmlandbacon it'd be good to have some clear principles to guide our template development. 
If anyone knows if these are documented, it'd be a great help.

#### Useful
GDS do have this https://github.com/alphagov/govuk-frontend/blob/main/docs/contributing/coding-standards/nunjucks-api.md 

### Questions
- What's missing here for you? 
- Any other suggestions?

**Comment by jonhurrell** [comment]
If you feel the GDS guidance is good enough, can you add an emoji to this comment (cheap poll ;))

**Comment by htmlandbacon** [comment, marked as answer]
I think this suitability covers the input, the other thee things that came to mind based on what I've seen:

1. Making sure text can be overridden
2. ability to add styles, attributes
3. specificity and composition
 
I think the final two points will be driven out by teams feeding back about use cases.
